### DISCUSSION TITLE
Add "Warning" and "Tip" settings to custom fields

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#12821] [Ideas] [answered]
It'd be great if it was possible to add warning and tip texts to custom fields, in addition to the basic instruction text.

To be clear; here are the tip and warning texts I'm referring to (the basic instruction would render below the label and above the field; the tip and warnings render below the field):

<img width="689" alt="CleanShot 2023-03-07 at 18 24 29@2x" src="https://user-images.githubusercontent.com/298510/223500470-d1954943-ad17-4e63-9efd-334fe8759d6d.png">

(I realise that there are UI elements for tips and warnings, but IMO field-specific warnings and tips still have merit.)

**Comment by mmikkel** [comment]
I'm not sure if these settings should be "Default Warning" and "Default Tip", and possible to "relabel" in field layouts (i.e. similar to the "Default Instructions"). Possibly yes, due to consistency if nothing else?

**Reply by brandonkelly** [reply]
Yeah, probably, along with “Hide” checkboxes in case you want to explicitly remove the default tip in some cases.

**Comment by mmikkel** [comment, marked as answer]
Added in 5.4.0 (https://github.com/craftcms/cms/discussions/15632#discussioncomment-10503113) 🎉
